hw4/hw4.py

```diff
@@ -192,19 +163,29 @@ def cat_resp_cont_predictor(df, i, response, predictors, group_labels):
     return
 
 
-result_response = check_response(df, response)
+if __name__ == "__main__":
+    diabetes = datasets.load_diabetes(as_frame=True)
+    df = diabetes["frame"]
+    predictors = ["age", "sex", "bmi", "bp", "s1", "s2", "s3", "s4", "s5", "s6"]
+    response = ["target"]
 
-print(result_response)
-for i in range(0, len(df.columns) - 1):
-    # result_response = check_response(df,i,response)
-    result_predictor = check_predictor(df, i, predictors)
-    if result_response == "CAT_RES":
-        if result_predictor == "CAT_PRED":
-            print("cat_res_cat_pred()")
-        else:
-            cat_resp_cont_predictor(df, i, response, predictors, group_labels)
-    else:
-        if result_predictor == "CAT_PRED":
-            print("cont_res_cat_pred()")
+    # variable_type(df,predictors,response)
+    lr(df, predictors, response)
+    random_forest()
+    result_response = check_response(df, response)
+
+    print(result_response)
+    for i in range(0, len(df.columns) - 1):
+        # result_response = check_response(df,i,response)
+        result_predictor = check_predictor(df, i, predictors)
+        if result_response == "CAT_RES":
+            if result_predictor == "CAT_PRED":
+                print("cat_res_cat_pred()")
+            else:
+                print("Hi")
+        #               cat_resp_cont_predictor(df, i, response, predictors, group_labels)
         else:
```

Comment:
looks great, but I cant find your code for the tests, and putting results in a table